include/gc/Transforms/Passes.td

```diff
@@ -93,6 +93,20 @@ def LinalgToXeGPU : Pass<"linalg-to-xegpu", "func::FuncOp"> {
                "DPAS register block sizes MxNxK">,
   ];
 }
+
+def AddContextArg : Pass<"add-ctx-arg", "func::FuncOp"> {
```

Comment:
Should it be applied to kernels only?

Reply:
No, it should be applied to the functions, that launch the kernels. The argument is passed to the runtime.